src/huggingface_hub/hf_api.py

```diff
@@ -3581,13 +3584,16 @@ def update_repo_settings(
         self,
         repo_id: str,
         *,
-        gated: Literal["auto", "manual", False] = False,
+        gated: Optional[Literal["auto", "manual", False]] = False,
+        private: Optional[bool] = False,
```

Comment:
I think the default values for both `gated` and `private` attributes should be set to `None` instead